core/util.go

```diff
@@ -211,3 +211,48 @@ func cloneMap[K comparable, T any](src map[K]T) map[K]T {
 	}
 	return dst
 }
+
+func envToMap(src []string) map[string]string {
+	dst := map[string]string{}
+
+	for _, e := range src {
+		env := strings.Split(e, "=")
+		dst[env[0]] = env[1]
+	}
+
+	return dst
+}
```

Comment:
In Go this conversion loses the order. You can see how the `dockerfile2llb` project does it:

https://github.com/moby/buildkit/blob/c563db65a089b31c95f7fab5f8139ba1526c37b5/frontend/dockerfile/dockerfile2llb/convert.go#L1408-L1432